### PR TITLE
Fix nat gateway type of LoadBalancer

### PR DESCRIPTION
**Pull request by niuzhenguo**
The url of NAT already updated to v2 on Cloud side, the v2.0 one doesn't work any more.

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/cloudprovider/huaweicloud/nat_connections.go [hunk 1]
```diff
@@ -225,7 +225,7 @@ func NewNATClient(natEndpoint, vpcEndpoint, id, accessKey, secretKey, securityTo
  *    >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
  */
 func (nat *NATClient) GetNATGateway(natGatewayId string) (*NATGateway, error) {
-	url := "/v2.0/nat_gateways/" + natGatewayId
+	url := "/v2/nat_gateways/" + natGatewayId
```
Comment (chengxiangdong) [inline pkg/cloudprovider/huaweicloud/nat_connections.go:228]: `projectID` needs to be specified for APIs of the new version.

File: pkg/cloudprovider/huaweicloud/nat_connections.go [hunk 2]
```diff
@@ -337,7 +337,7 @@ func (nat *NATClient) GetDNATRule(dnatRuleId string) (*DNATRule, error) {
 }
 
 func (nat *NATClient) ListDNATRules(params map[string]string) (*DNATRuleList, error) {
-	url := "/v2.0/dnat_rules"
+	url := "/v2/dnat_rules"
```
Comment (chengxiangdong) [inline pkg/cloudprovider/huaweicloud/nat_connections.go:340]: ditto

File: pkg/cloudprovider/huaweicloud/nat_connections.go [hunk 3]
```diff
@@ -297,7 +297,7 @@ func (nat *NATClient) CreateDNATRule(dnatRuleConf *DNATRule) (*DNATRule, error)
 }
 
 func (nat *NATClient) DeleteDNATRule(dnatRuleId string) error {
-	url := "/v2.0/dnat_rules/" + dnatRuleId
+	url := "/v2/dnat_rules/" + dnatRuleId
```
Comment (chengxiangdong) [inline pkg/cloudprovider/huaweicloud/nat_connections.go:300]: we should use the following API:
`DELETE /v2/{project_id}/nat_gateways/{nat_gateway_id}/dnat_rules/{dnat_rule_id}`

File: pkg/cloudprovider/huaweicloud/nat_connections.go [hunk 4]
```diff
@@ -315,7 +315,7 @@ func (nat *NATClient) DeleteDNATRule(dnatRuleId string) error {
 }
 
 func (nat *NATClient) GetDNATRule(dnatRuleId string) (*DNATRule, error) {
-	url := "/v2.0/dnat_rules/" + dnatRuleId
+	url := "/v2/dnat_rules/" + dnatRuleId
```
Comment (chengxiangdong) [inline pkg/cloudprovider/huaweicloud/nat_connections.go:318]: we should use the following API:
`/v2/{project_id}/dnat_rules/{dnat_rule_id}`

File: pkg/cloudprovider/huaweicloud/nat_connections.go (hunk 1 above)
Comment (niuzhenguo) [reply pkg/cloudprovider/huaweicloud/nat_connections.go]: done

**Comment by chengxiangdong** [comment]
/lgtm
The APIs has been upgraded, we still need an E2E test to test the `DNAT` service, can you add it?

**Comment by niuzhenguo** [comment]
> /lgtm The APIs has been upgraded, we still need an E2E test to test the `DNAT` service, can you add it?

sure, a new issue has been added to track this #134

**Pull request merged**